planner/src/planner.cpp

```diff
@@ -0,0 +1,333 @@
+#include <planner/planner.hpp>
+
+#define sq(x) (x) * (x)
+#define echo(X) std::cout << X << std::endl
+int p = 0;
+double yaw_change = 0;
+
+namespace msm = boost::msm;
+typedef msm::back::state_machine<ariitk::planner::fsm> fsm_;
+static char const* const state_names[] = {"Rest", "Hover", "Before", "After"};
+
+namespace ariitk::planner {
+
+bool fsm::verbose = true;
+void fsm::init(ros::NodeHandle& nh) {
+	rough_pose_[0][0] = 5.0;
+	rough_pose_[0][1] = 0.0;
+	rough_pose_[0][2] = 2.7;
+	rough_pose_[1][0] = 8.0;
+	rough_pose_[1][1] = -6.0;
+	rough_pose_[1][2] = 2.7;
+	rough_pose_[2][0] = 4.0;
+	rough_pose_[2][1] = -9.0;
```

Comment:
best parametrize these coordinates, it's possible to take a list of lists as a ROS parameter